test/export/test_export.py

```diff
@@ -6277,6 +6213,39 @@ def forward(self, x, y):
             ep.graph_module.code
         )
 
+    def test_slice_with_floordiv(self):
+        # slice operation emits runtime assert s0//2 <= s1
+        class M1(torch.nn.Module):
+            def forward(self, x, y):
+                d = x.size(0) // 2
+                return y[d:]
+
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.m1 = M1()
+
+            def forward(self, x, y):
+                d = x.size(0) // 2
+                m1_res = self.m1(x, y)
+                return y[d:] + m1_res
+
+        inputs = (torch.ones(10), torch.ones(10))
+        d0 = torch.export.Dim("d0", max=2048)
+        d1 = torch.export.Dim("d1", max=2048)
+        ep = export(
+            M(),
+            inputs,
+            dynamic_shapes=((d0,), (d1,)),
+        )
+        ep.module()(torch.ones(8), torch.ones(4))
+        ep.module()(torch.ones(8), torch.ones(5))
+        with self.assertRaisesRegex(
+            RuntimeError,
+            r"Runtime assertion failed for expression \(s0//2\) \<\= s1",
```

Comment:
Cool! As a TODO, maybe we should try renaming these symbols to input shapes in the exported program (i.e., build "sources" from the placeholder names and their symbolic shapes). So here we'd get something like `x.shape[0] // 2 <= x.shape[1]`.

Note that I'm not suggesting reusing the shape env sources, but rather build one from (your now nicely named) placeholder names.

Reply:
Hmm do you mean just prettifying the runtime assert messages, or at some level renaming the symbols? I get the first option, but maybe I'm missing something smarter.

This could be a good BE task for next week?